### ISSUE TITLE
Critical/Glancing/Maimed/etc feature & chat messages.

### ISSUE THREAD

**Issue by Formula350** [mcmonkeyprojects/Sentinel#209]
That was one of the best parts of Sentry in my opinion, and added a very nice RPG element to combat. I was quite bummed out to find that had been removed. Would be greatly appreciated if it could make a comeback! 👍

**Comment by mcmonkey4eva** [comment]
Could you describe the feature you're requesting?

**Comment by Formula350** [comment]
This, from Sentry (these are from the config.yml):
````yaml
#Default options for new Sentries
DefaultOptions:
 Drops: false
 KillDrops: true
 Criticals: true    <------------------
 Invincible: false
 Retaliate: true
 FriendlyFire: false
 Targetable: true
````
The Chance that they happen...
````yaml
#These are the percentage chancees of doing a certain damage 
#to a Sentry if Criticals is turned on. Whole numbers only. 
#Remove or Set to 0 to disable a hit type.
HitChances:
 Miss: 4
 Glance: 4
 Crit1: 6
 Crit2: 4
 Crit3: 2
````
And the accompany chat messages for when they do occur *(although, `"Hit"` always displayed,  and `"Block"` would display when the Sentry's armor level was higher than the damage delt by a player's weapon)*...
````yaml
#Configurable messages used by all Sentries. 
#You can use <NPC>, <PLAYER>, <ITEM>, and <AMOUNT> as placeholders. 
#Remove a message or set it to "" to prevent it from displaying at all.
GlobalTexts:
 Miss: "&7*** You MISS <NPC>."
 Hit: "&f*** You hit <NPC> for <AMOUNT> damage."
 Crit1: "&e*** You injure <NPC> for <AMOUNT> damage."
 Crit2: "&6*** You MAIM <NPC> with your <ITEM> for <AMOUNT> damage."
 Crit3: "&c*** You DISEMBOWEL <NPC> with your <ITEM> for <AMOUNT> damage."
 Block: "&7*** You fail to penetrate <NPC>'s armor."
 Glance: "&7*** Your blow glances off <NPC> for <AMOUNT> damage."
````

I wasn't able to find anything via searching for critical or chance in your Sentry fork's code *(which... I was using up until the move to MC1.11.2 this week)*, but **did** manage to in the original repo's code, if that helps:
https://github.com/jrbudda/Sentry/search?q=chance&type=

The build I had been using in 1.8 was Sentry 1.8.5 b28 (from the Citizens Jenkins server).

**Comment by mcmonkey4eva** [comment]
That seems like the job for a separate but compatible RPG plugin or script moreso than a job for Sentinel, wouldn't you say?

**Comment by Formula350** [comment]
heh I wouldn't be able to say, as I'm not the dev of this plugin :) Thus my reasoning for making the request. If you feel it is not befitting of Sentinel, I cannot argue that decision, nor would I as it's not my place as an user who can't even code lol

My hope, albeit driven by my coding ignorance, was that due to being formally in Sentry, that porting the code wouldn't be too time consuming or much of a challenge.  

Either way, I appreciate the time and effort put in to Sentinel (and Sentry, and Citizens)!
Thanks
-Form

**Comment by mcmonkey4eva** [comment]
I don't think that particularly belongs in Sentinel.

Sentry code was a wreck, and Sentinel includes no Sentry code at all.
I rebuilt the intent of Sentry, not just copying the code. I tried to maintain Sentry's original code (written by auf and jrbudda) for so long before I gave up due to it just being so full of bugs and weird decisions.

I might be able to still add this, or add tooling for user creation (EG events that can be scripted using Denizen, or similar)
